dGame/dComponents/TriggerComponent.cpp

```diff
@@ -137,51 +331,72 @@ void TriggerComponent::HandleSetPhysicsVolume(Entity* targetEntity, std::vector<
 		GeneralUtils::TryParse<float>(argArray.at(4), direction.z);
-		phanPhys->SetDirection(direction);
+		phantomPhysicsComponent->SetDirection(direction);
 	}
 	if (argArray.size() > 5) {
 		uint32_t min;
 		GeneralUtils::TryParse<uint32_t>(argArray.at(6), min);
-		phanPhys->SetMin(min);
+		phantomPhysicsComponent->SetMin(min);
 
 		uint32_t max;
 		GeneralUtils::TryParse<uint32_t>(argArray.at(7), max);
-		phanPhys->SetMax(max);
+		phantomPhysicsComponent->SetMax(max);
 	}
 
-	// TODO: why is this contruct and not serialize?
-	if (target == "self") EntityManager::Instance()->ConstructEntity(m_Parent);
+	EntityManager::Instance()->SerializeEntity(targetEntity);
 }
 
-void TriggerComponent::HandleUpdateMission(Entity* targetEntity, std::vector<std::string> argArray) {
-	CDMissionTasksTable* missionTasksTable = CDClientManager::Instance()->GetTable<CDMissionTasksTable>("MissionTasks");
-	std::vector<CDMissionTasks> missionTasks = missionTasksTable->Query([=](CDMissionTasks entry) {
-		std::string lowerTargetGroup;
-		for (char character : entry.targetGroup) {
-			lowerTargetGroup.push_back(std::tolower(character)); // make lowercase to ensure it works
-		}
+void TriggerComponent::HandleSetPhysicsVolumeStatus(Entity* targetEntity, std::string args) {
+	PhantomPhysicsComponent* phanPhys = targetEntity->GetComponent<PhantomPhysicsComponent>();
+	if (!phanPhys) return;
+	phanPhys->SetPhysicsEffectActive(args == "On");
+	EntityManager::Instance()->SerializeEntity(targetEntity);
+}
 
-		return (lowerTargetGroup == argArray[4]);
-	});
+void TriggerComponent::HandleActivateSpawnerNetwork(std::string args){
+	for (auto* spawner : dZoneManager::Instance()->GetSpawnersByName(args)) {
+		spawner->Activate();
+	}
+}
 
-	for (const CDMissionTasks& task : missionTasks) {
-		MissionComponent* missionComponent = targetEntity->GetComponent<MissionComponent>();
-		if (!missionComponent) continue;
+void TriggerComponent::HandleDeactivateSpawnerNetwork(std::string args){
+	for (auto* spawner : dZoneManager::Instance()->GetSpawnersByName(args)) {
+		spawner->Deactivate();
+	}
+}
 
-		missionComponent->ForceProgress(task.id, task.uid, std::stoi(argArray[2]));
+void TriggerComponent::HandleResetSpawnerNetwork(std::string args){
+	for (auto* spawner : dZoneManager::Instance()->GetSpawnersByName(args)) {
+		spawner->Reset();
 	}
 }
 
-void TriggerComponent::HandleFireEvent(Entity* targetEntity, std::string args) {
-	for (CppScripts::Script* script : CppScripts::GetEntityScripts(targetEntity)) {
-		script->OnFireEventServerSide(targetEntity, m_Parent, args, 0, 0, 0);
+void TriggerComponent::HandleDestroySpawnerNetworkObjects(std::string args){
+	for (auto* spawner : dZoneManager::Instance()->GetSpawnersByName(args)) {
+		for (auto* node : spawner->m_Info.nodes) {
+			for (const auto& element : node->entities) {
+				auto* entity = EntityManager::Instance()->GetEntity(element);
+				if (entity == nullptr) continue;
+				entity->Kill();
+			}
+			node->entities.clear();
+		}
```

Comment:
Could we not just call Spawner::Reset here?

Reply:
I would say no, since reset does more than that, and there's specifically a reset, vs kill all network objects

Reply:
It may make more sense to make this functionality into a function then so we aren't repeating ourself with the same loop in two places